colossalai/kernel/extensions/cpu_adam/arm_extension.py

```diff
@@ -6,15 +6,15 @@ class ArmCPUAdamExtension(BaseExtension):
     def __init__(self) -> None:
         super().__init__()
         self.kernel_builder = ArmCPUAdamBuilder()
-        self._requires_build = False
+        self._requires_build = True
 
```

Comment:
You should consider the necessity to build between runs. You should have something like

```bash
def requires_build(self):
      # you should check whether the kernel has been built here

```

Reply:
The build() in extension_builder.py has already done the check.  
![image](https://github.com/hpcaitech/ColossalAI/assets/32676639/a495856c-fa10-4994-a85b-c1ca7d9756d0)
![image](https://github.com/hpcaitech/ColossalAI/assets/32676639/74af8963-262a-4baf-895c-b86d6138c463)